skymp5-server/cpp/server_guest_lib/PapyrusObjectReference.cpp

```diff
@@ -273,3 +273,61 @@ VarValue PapyrusObjectReference::Activate(
   }
   return VarValue::None();
 }
+
+VarValue PapyrusObjectReference::GetPositionX(
+  VarValue self, const std::vector<VarValue>& arguments)
+{
+  if (auto selfRefr = GetFormPtr<MpObjectReference>(self)) {
+    return VarValue(selfRefr->GetPos().x);
+  }
+  return VarValue::None();
+}
+
+VarValue PapyrusObjectReference::GetPositionY(
+  VarValue self, const std::vector<VarValue>& arguments)
+{
+  if (auto selfRefr = GetFormPtr<MpObjectReference>(self)) {
+    return VarValue(selfRefr->GetPos().y);
+  }
+  return VarValue::None();
+}
+
+VarValue PapyrusObjectReference::GetPositionZ(
+  VarValue self, const std::vector<VarValue>& arguments)
+{
+  if (auto selfRefr = GetFormPtr<MpObjectReference>(self)) {
+    return VarValue(selfRefr->GetPos().z);
+  }
+  return VarValue::None();
+}
+
+VarValue PapyrusObjectReference::SetPosition(
+  VarValue self, const std::vector<VarValue>& arguments)
+{
+  if (auto selfRefr = GetFormPtr<MpObjectReference>(self)) {
+    if (arguments.size() < 3) {
+      throw std::runtime_error(
+        "SetPosition requires at least three arguments");
+    }
+    // TODO: Add movement/position changing sync for ingame objects and remove
+    // workaround below. Don't forget that SetPosition has fading effect. Also
+    // note that this change is global. So players should get new position on
+    // entering area or even outside of it
+    selfRefr->SetPosAndAngleSilent(
+      NiPoint3((float)static_cast<double>(arguments[0].CastToFloat()),
```

Comment:
C-style cast after a C++ cast? Pleaseleave C++ one only...

Reply:
That was copypaste from SetAngle